delivery/services/staging_service.py

```diff
@@ -1,215 +0,0 @@
-        """
-        Copies the file or directory indicated by the staging order by calling the external_program_service.
-        It will attempt the copying and update the database with the status of the StagingOrder depending on the
-        outcome.
-        :param staging_order_id: The id of the staging order to execute
-        :param external_program_service: A instance of ExternalProgramService
-        :param session_factory: A factory method which can produce a new sql alchemy Session instance
-        :param staging_repo: A instance of DatabaseBasedStagingRepository
-        :return: None, only reports back through side-effects
-        """
-        session = session_factory()
-
-        # This is a somewhat hacky work-around to the problem that objects created in one
-        # thread, and thus associated with another session cannot be accessed by another
-        # thread, there fore it is re-materialized in here...
-        staging_order = staging_repo.get_staging_order_by_id(staging_order_id, session)
-        try:
-
-            cmd = ['rsync', '-r', staging_order.source, staging_order.staging_target]
-
-            execution = external_program_service.run(cmd)
-
-            staging_order.pid = execution.pid
-            session.commit()
-
-            execution_result = external_program_service.wait_for_execution(execution)
-
-            if execution_result.status_code == 0:
-                staging_order.status = StagingStatus.staging_successful
-                session.commit()
-                log.info("Successfully staged: {}".format(staging_order))
-            else:
-                staging_order.status = StagingStatus.staging_failed
-                session.commit()
-                log.info("Failed in staging: {} because rsync returned exit code: {}".
-                         format(staging_order, execution_result.status_code))
-
-        # TODO Better exception handling here...
-        except Exception as e:
-            staging_order.status = StagingStatus.staging_failed
-            log.info("Failed in staging: {} because this exception was logged: {}".
-                     format(staging_order, e))
-        finally:
-            # Always commit the state change to the database
-            session.commit()
-
-    def stage_order(self, stage_order):
-        """
-        Validate a staging order and hand of the actual stating to a separate thread.
-        :param stage_order: to stage
-        :return: None
-        """
-
-        session = self.session_factory()
-
-        try:
-
-            if stage_order.status != StagingStatus.pending:
-                raise InvalidStatusException("Cannot start staging a delivery order with status: {}".
-                                             format(stage_order.status))
-
-            stage_order.status = StagingStatus.staging_in_progress
-            session.commit()
-
-            thread = threading.Thread(target=StagingService._copy_dir,
```

Comment:
Great suggestion! I've implemented this now and it seems to be working well. :smile: